src/python/pants/engine/internals/graph.py

```diff
@@ -1300,6 +1304,13 @@ async def resolve_dependencies(
         WrappedTargetRequest(request.field.address, description_of_origin="<infallible>"),
     )
     tgt = wrapped_tgt.target
+
+    # This predicate allows the dep graph to ignore dependencies of selected targets
+    # including any explicit deps and any inferred deps.
+    # For example, to avoid traversing the deps of package targets.
+    if not request.should_resolve_deps_predicate(tgt, request.field):
+        return Addresses([])
```

Comment:
@stuhood What do you think of pushing the predicate evaluation into the `resolve_dependencies` rule via `DependenciesRequest`? This is a standardized predicate that can be passed via `DependenciesRequest`, 
 `CoarsenedTargetsRequest`, or `TransitiveTargetsRequest`.